src/components/navigation/index.tsx

```diff
@@ -55,12 +92,40 @@ function MainNavigation({ maxWidth }: { maxWidth: ContainerProps['maxWidth'] })
             </Link>
 
             <IconButton
+              aria-describedby={id}
               aria-label="global-settings"
+              onClick={handleClick}
               sx={{ mr: 4, color: 'secondary.main', padding: 1 }}
             >
               <LanguageIcon sx={{ fontSize: '1.5rem' }} />
             </IconButton>
-
+            <Popover
+              id={id}
+              open={open}
+              anchorEl={anchorEl}
+              onClose={handleClose}
+              anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+              }}
+            >
+              <Stack>
+                {LANGUAGE_LIST.map((lang) => (
+                  <MenuItem
```

Comment:
create component LanguageMenuItem as wrapper on MenuItem and pass handleChange function and lang.code inside there as props. Otherwise useCallback doesn't make sense. you create new function for each menu item and for each render
() => {
                      handleChange({
                        target: { value: lang.code },
                      } as SelectChangeEvent<string>);
                      handleLanguageMenuClose();
                    }

for